src/interactive/azext_interactive/azclishell/scenario_suggest.py

```diff
@@ -32,12 +40,44 @@ def update(self, sample: str):
                 self.param_value_map[cur_param] += ' ' + part
                 self.param_value_map[cur_param] = self.param_value_map[cur_param].strip()
 
-    def get_suggestion(self, cli, buffer, document):
+    def update_customized_param_value_map(self, command_text: str):
+        """Update the value of a parameter in the customized parameter value map"""
+        # remove the 'az ' part of command
+        command_text = command_text.split('az ')[-1]
+        cur_param = ''
+        special_global_param_list = None
+        for part in command_text.split():
+            if part.startswith('-'):
+                # if the parameter is a special global parameter, use the parameter itself as the key
+                if part in self.special_global_param_map.keys():
+                    # Because '--g' and '--resource-group' are the same parameter, we need to both support in the customized map
+                    special_global_param_list = self.special_global_param_map[part]
+                    for param in special_global_param_list:
+                        self.customized_param_value_map[param] = ''
+                    cur_param = ''
+                else:
+                    cur_param = self.param_value_map[part]
+                    self.customized_param_value_map[cur_param] = ''
+                    special_global_param_list = None
+            elif cur_param:
+                self.customized_param_value_map[cur_param] += ' ' + part
```

Comment:
Could you please add some comments to explain the meaning of `param_value_map` and `customized_param_value_map`? Such as, give some examples for their key and value respectively

Reply:
Already modified with the new commit by optimizing param name and add notes: https://github.com/wu5bocheng/azure-cli-extensions/pull/2/commits/0169eb35aef4ebc0980a79ec1a258244f1deeb8e